File: src/main/java/com/codeperfection/shipit/dto/common/PaginationFilterDto.java
package com.codeperfection.shipit.dto.common;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.Max;
import javax.validation.constraints.Min;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class PaginationFilterDto {

    @Min(0)
    private Integer page = 0;

    @Min(1)
    @Max(100)
    private Integer size = 20;
}

File: src/main/kotlin/com/codeperfection/shipit/dto/common/PaginationFilterDto.kt
package com.codeperfection.shipit.dto.common

import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min
import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Sort

data class PaginationFilterDto(
    @field:Min(0)
    val page: Int = 0,
    @field:Min(1)
    @field:Max(100)
    val size: Int = 100
) {
    fun toPageable(): Pageable = PageRequest.of(page, size, Sort.by("createdAt"))
}
